controllers/placementapi_controller.go

```diff
@@ -491,19 +492,14 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return result, err
 	}
 
-	// Only expose the service is the deployment succeeded
-	if !instance.Status.Conditions.IsTrue(condition.DeploymentReadyCondition) {
-		Log.Info("Waiting for the Deployment to become Ready before exposing the sevice in Keystone")
-		return ctrl.Result{}, nil
-	}
-
-	// remove finalizers from unused MariaDBAccount records
-	err = mariadbv1.DeleteUnusedMariaDBAccountFinalizers(ctx, h, placement.DatabaseName, instance.Spec.DatabaseAccount, instance.Namespace)
-	if err != nil {
-		return ctrl.Result{}, err
+	if instance.Status.ReadyCount > 0 || *instance.Spec.Replicas == 0 {
```

Comment:
Why do you switched from checking the DeploymentReady condition to checking the ReadyCount instead? I see both being set now under the observedGeneration condition. 
//later

I think I see. So at this point the DeploymentReady is Unknown as it is reset at the start and only set below. But we are not resetting the ReadyCount. It think instead of relying on the leaky ReadyCount we should move the mariadbaccunt cleanup to be executed after the Deployment is reconciled.

Reply:
Sorry my bad, need to get back to this